store/store_download.go

```diff
@@ -253,6 +259,109 @@ func downloadReqOpts(storeURL *url.URL, cdnHeader string, opts *DownloadOptions)
 	return &reqOptions
 }
 
+type downloadTimeoutError struct {
+	// for testing, to inspect speed
+	Speed float64
+}
+
+func (e *downloadTimeoutError) Error() string {
+	return fmt.Sprintf("download timed out")
```

Comment:
I think we could use more elaborate error message here, eg.
```
fmt.Sprintf("download too slow: %.2f bytes/sec", e.Speed)
```

Reply:
good idea